app/src/main/java/org/mozilla/fenix/settings/DeleteBrowsingDataOnQuitFragment.kt

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.settings
+
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.preference.CheckBoxPreference
+import androidx.preference.Preference
+import androidx.preference.PreferenceFragmentCompat
+import androidx.preference.SwitchPreference
+import org.mozilla.fenix.R
+import org.mozilla.fenix.ext.getPreferenceKey
+import org.mozilla.fenix.utils.Settings
+
+class DeleteBrowsingDataOnQuitFragment : PreferenceFragmentCompat() {
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        setPreferencesFromResource(R.xml.delete_browsing_data_quit_preferences, rootKey)
+    }
+
+    @Suppress("ComplexMethod")
+    override fun onResume() {
+        super.onResume()
+        activity?.title = getString(R.string.preferences_delete_browsing_data_on_quit)
+        (activity as AppCompatActivity).supportActionBar?.show()
+
+        class CheckboxUpdater : SharedPreferenceUpdater() {
```

Comment:
Creating an actual class inside of this method doesn't look right. Could we just create the updater inline

```
val checkboxUpdater = object : SharedPreferenceUpdater() {
...
}
```
and then use that as the listener?

Reply:
wow yep i knew this looked weird lol thanks!